src/Inciter/AMR/refinement.h

```diff
@@ -0,0 +1,779 @@
+#ifndef AMR_refinement_h
+#define AMR_refinement_h
+
+#include <algorithm>
+
+#include "tet_store.h"
+#include "node_connectivity.h"
+
+// TODO: make this have a base class to support multiple generator schemes
+// using the policy design pattern
+
+namespace AMR {
+
+    class refinement_t {
+        private:
+
+            const size_t DEFAULT_REFINEMENT_LEVEL = 0; //TODO: Is this in the right place?
```

Comment:
We will read this out from user input. I will help (or will do it) in a second pass.